bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/__init__.py

```diff
@@ -61,7 +61,21 @@
 sys.modules['libcppyy'] = sys.modules[libcppyy_mod_name]
 
 # tell cppyy that libcppyy_backend is versioned
+def _check_py_version(lib_name, cbl_var):
+    import re
+    if re.match('^libcppyy_backend\d_\d$', lib_name):
+       # library name already has version
+       if lib_name.endswith(py_version_str):
+           return ''
+       else:
+           raise RuntimeError('CPPYY_BACKEND_LIBRARY variable ({})'
```

Comment:
I'm not sure I understand, once we raise the error due to the mismatch between the cppy_backend variable and the python version used what do we do? we just exit and then it's up to the user understanding that a different python needs to be used?

Reply:
In the case where the user specified `CPPYY_BACKEND_LIBRARY`, we check if the library has a version that conflicts with the current Python version we are running on. If so, we error out with an informative message to the user about this situation. I think it is better to stop and tell the user that the their environment has a conflict with the Python they are running.